bin/Cargo.toml

```diff
@@ -37,6 +37,7 @@ recursor = ["hickory-server/recursor"]
 # Recursive Resolution is Experimental!
 resolver = ["hickory-server/resolver"]
 sqlite = ["hickory-server/sqlite", "dep:rusqlite"]
+prometheus-metrics = ["hickory-server/metrics", "__prometheus"]
```

Comment:
Why did you add the indirection between `prometheus-metrics` and `__prometheus`?

Reply:
I had seen that in the code all `#[cfg(feature = "")]` guards used values starting with `__` therefore I've added that.

Technically I I think there should be only a distinction between `metrics` and `prometheus` where the later depends on the first.

Reply:
That's because there are multiple features (like `dnssec-aws-lc-rs`, `dnssec-ring` -> `__dnssec`). That's not the case here.

Reply:
Thanks.

Reply:
Removed.